GitUI/UserControls/TextBoxSubmit.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace GitUI.UserControls
+{
+    /// <summary>
+    /// A TextBox that allows trapping the Enter key.
+    /// </summary>
+    public class TextBoxSubmit : TextBox
```

Comment:
It subclassing really necessary? `OnKeyDown` should suffice.

Reply:
It is. The Enter key is a special key, and is therefore consumed by the system rather than being sent to the event handler. I saw some older examples to use `OnKeyDown` or `OnKeyPress`, but testing revealed that code was broken in current .Net versions. Overriding the method intended for handling the Enter key is the only way that's guaranteed to not arbitrarily break on future .Net versions.